drivers/serial/Kconfig.stm32

```diff
@@ -25,15 +25,39 @@ config UART_STM32
 
 if UART_STM32
 
-config UART_STM32U5_ERRATA_DMAT
-	bool
-	default y
-	depends on SOC_STM32U575XX || SOC_STM32U585XX || \
+if SOC_STM32U575XX || SOC_STM32U585XX || \
 		   SOC_STM32H562XX || SOC_STM32H563XX || SOC_STM32H573XX
+
+choice UART_STM32U5_ERRATA_DMAT
+	prompt "Workaround for DMAT errata on selected devices"
+	default UART_STM32U5_ERRATA_DMAT_LOWPOWER if PM
+	default UART_STM32U5_ERRATA_DMAT_NOCLEAR if !PM
 	help
 	  Handles erratum "USART does not generate DMA requests after
 	  setting/clearing DMAT bit".
 	  Seen in Errata Sheet 0499 § 2.19.2 and §2.20.1 for stm32u57x/u58x,
 	          Errata Sheet 0565 § 2.14.1 and §2.15.1 for stm32h56x/h57x
 
-endif
+config UART_STM32U5_ERRATA_DMAT_NONE
+	bool "No workaround"
+	help
+	  No workaround selected or SoC not affected.
```

Comment:
Now that the choice is made specific to affected SoC, this specific symbol is likely not useful anymore. Did you had a try removing it and compile on non-affected SoC ?